sql/src/main/java/org/apache/druid/sql/calcite/run/SqlEngine.java

```diff
@@ -116,4 +122,29 @@ QueryMaker buildQueryMakerForInsert(
   default void initContextMap(Map<String, Object> contextMap)
   {
   }
+
+  /**
+   * Returns a {@link SqlStatementFactory} which uses this engine to create statements.
+   */
+  SqlStatementFactory getSqlStatementFactory();
+
+  /**
+   * Returns a list of {@link QueryInfo} containing the currently running queries using this engine.
+   */
+  default List<QueryInfo> getRunningQueries(
+      boolean selfOnly,
+      AuthenticationResult authenticationResult,
+      AuthorizationResult authorizationResult
+  )
+  {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Cancels a currently running query given the {@link PlannerContext} for the query.
+   */
+  default void cancelQuery(PlannerContext plannerContext, QueryScheduler queryScheduler)
+  {
+    throw new UnsupportedOperationException();
```

Comment:
This should be a `DruidException` with appropriate persona-oriented message. The persona should be `USER` and the message should say that the requested engine does not support cancellation.